packages/material-ui-utils/src/useResizeObserver.js

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import ResizeObserverPolyfill from 'resize-observer-polyfill';
+
+export default function useResizeObserver() {
+  const [rect, setRect] = React.useState(null);
+  const ref = React.useRef(null);
+  React.useEffect(() => {
+    let observer;
+    try {
+      observer = new ResizeObserver(([item]) => {
+        setRect(item);
+      });
+    } catch (err) {
+      observer = new ResizeObserverPolyfill(([item]) => {
+        setRect(item);
+      });
+    }
+    const item = ref.current;
```

Comment:
We have no guarantees that the ref will be resolved in time here. I don't think that the current API works. I would recommend a similar signature to https://material-ui.com/components/app-bar/#usescrolltrigger-options-trigger